dotCMS/src/integration-test/java/com/dotcms/content/elasticsearch/business/ESContentFactoryImplTest.java

```diff
@@ -363,4 +387,62 @@ public void testDeleteVersion_KeepPermissions() throws DotSecurityException, Dot
         }
     }
 
+    @Test
+    @UseDataProvider("testCases")
+    public void testIndexSearchFilteringByDates(final TestCase testCase){
+        final long languageId = APILocator.getLanguageAPI().getDefaultLanguage().getId();
+        final ContentType contentType = TestDataUtils.getNewsLikeContentType();
+
+        LocalDateTime today, yesterday, tomorrow;
+
+        final LocalDateTime now = LocalDateTime.now();
+
+        if (testCase.formatPattern.equals("hh:mm:ssa") || testCase.formatPattern.equals("hh:mm:ss a") ||
+             testCase.formatPattern.equals("HH:mm:ss") || testCase.formatPattern.equals("hh:mma") ||
+             testCase.formatPattern.equals("hh:mm a") || testCase.formatPattern.equals("HH:mm")){
+            today = LocalDateTime.of(1970, 1, 1, now.getHour(), now.getMinute(), now.getSecond());
+            yesterday = LocalDateTime.of(1970, 1, 1, 0, 0, 0);
+            tomorrow = LocalDateTime.of(1970, 1, 1, 23, 59, 59);
+        } else{
+            today = now;
+            yesterday = today.minusDays(1).withHour(0).withMinute(0).withSecond(0);
+            tomorrow = today.plusDays(1).withHour(23).withMinute(59).withSecond(59);
+        }
+
+
+        Contentlet contentlet = TestDataUtils.getNewsContent(false, languageId, contentType.id());
+        contentlet.setDateProperty("sysPublishDate", Date.from(today.atZone(ZoneId.systemDefault()).toInstant()));
+
+        contentlet = ContentletDataGen.checkin(contentlet, IndexPolicy.WAIT_FOR);
+
+        final DateTimeFormatter myFormatObj = DateTimeFormatter.ofPattern(testCase.formatPattern);
+
+        //Test date ranges with TO (upper case)
+        String query = "+contentType:" + contentType.variable() + " +" + contentType.variable()
+                + ".sysPublishDate:[" + yesterday.format(myFormatObj) + " TO " + tomorrow
+                .format(myFormatObj) + "]";
+
+        SearchHits searchHits = instance.indexSearch(query,-1, -1, null);
+        validateQueryResults(contentlet, searchHits);
```

Comment:
maybe we can have a private method here 
https://github.com/dotCMS/core/pull/17650/files#diff-159aa097d6eb5735fbb174b8571d3c9cR421-R426

, and the reuse it here

https://github.com/dotCMS/core/pull/17650/files#diff-159aa097d6eb5735fbb174b8571d3c9cR429